google/cloud/security/common/gcp_api/admin_directory.py

```diff
@@ -13,70 +13,135 @@
 # limitations under the License.
 
 """Wrapper for Admin Directory  API client."""
+from googleapiclient import errors
+from httplib2 import HttpLib2Error
 
-from oauth2client.service_account import ServiceAccountCredentials
-from ratelimiter import RateLimiter
-
-from google.cloud.security.common.gcp_api import _base_client
+from google.cloud.security.common.gcp_api import _base_repository
 from google.cloud.security.common.gcp_api import errors as api_errors
+from google.cloud.security.common.util import log_util
 
+LOGGER = log_util.get_logger(__name__)
 
-class AdminDirectoryClient(_base_client.BaseClient):
-    """GSuite Admin Directory API Client."""
 
-    API_NAME = 'admin'
-    DEFAULT_QUOTA_TIMESPAN_PER_SECONDS = 86400 # pylint: disable=invalid-name
-    REQUIRED_SCOPES = frozenset([
-        'https://www.googleapis.com/auth/admin.directory.group.readonly'
-    ])
+class AdminDirectoryRepository(_base_repository.BaseRepositoryClient):
```

Comment:
I feel that names of the derived and base classes are misaligned. I would expect the name of this class to be AdminDirectoryRepositoryClient.

Reply:
Good call, fixing.